src/libs/actions/App.js

```diff
@@ -214,23 +261,25 @@ function setUpPoliciesAndNavigate(session) {
     const makeMeAdmin = url.searchParams.get('makeMeAdmin');
     const policyName = url.searchParams.get('policyName');
 
-    // Sign out the current user if we're transitioning from oldDot with a different user
-    const isTransitioningFromOldDot = Str.startsWith(url.pathname, Str.normalizeUrl(ROUTES.TRANSITION_FROM_OLD_DOT));
-    if (isLoggingInAsNewUser && isTransitioningFromOldDot) {
+    // Sign out the current user if we're transitioning with a different user
+    const isTransitioning = Str.startsWith(url.pathname, Str.normalizeUrl(ROUTES.TRANSITION_BETWEEN_APPS));
+    if (isLoggingInAsNewUser && isTransitioning) {
         Session.signOut();
     }
 
-    const shouldCreateFreePolicy = !isLoggingInAsNewUser && isTransitioningFromOldDot && exitTo === ROUTES.WORKSPACE_NEW;
+    const shouldCreateFreePolicy = !isLoggingInAsNewUser && isTransitioning && exitTo === ROUTES.WORKSPACE_NEW;
     if (shouldCreateFreePolicy) {
-        Policy.createWorkspace(policyOwnerEmail, makeMeAdmin, policyName, true);
+        createWorkspaceAndNavigateToIt(policyOwnerEmail, makeMeAdmin, policyName, true);
         return;
     }
     if (!isLoggingInAsNewUser && exitTo) {
-        Navigation.isNavigationReady().then(() => {
-            // We must call goBack() to remove the /transition route from history
-            Navigation.goBack();
-            Navigation.navigate(exitTo);
-        });
+        Navigation.isNavigationReady()
+            .then(() => {
+                // We must call goBack() to remove the /transition route from history
+                Navigation.goBack();
+                Navigation.navigate(exitTo);
+            })
+            .then(endSignOnTransition);
```

Comment:
We also need to call `endSignOnTransition` when exitTo param isn't available to avoid [issue](https://github.com/Expensify/App/issues/52450)